### PR TITLE
Githubactions asg

### PR DESCRIPTION
**Pull request by NishaAnilBidave**
All the suggested changes have been made, kindly review and let me know if any changes are required.

### CODE REVIEW

File: autoscaling.tf [hunk 1]
```diff
@@ -0,0 +1,36 @@
+resource "aws_launch_configuration" "asg_launch" {
+  name_prefix   = "asg-launch"
+  image_id      = "ami-068257025f72f470d"
+  instance_type = "t3.small"
+  # user_data       = file("user_data.sh")
```
Comment (pasc-ed) [inline autoscaling.tf]: Why is this line commented ? If there's no need for it, it should be removed completely.

File: autoscaling.tf [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  security_groups = [aws_security_group.lb_sg.id, aws_security_group.bastion_host_server_sg.id]
```
Comment (pasc-ed) [inline autoscaling.tf]: Why is the launch configuration of the ASG using the SG of the loadbalancer and the bastion ? What is the reason behind this ?

File: bastion_host.tf [hunk 3]
```diff
@@ -0,0 +1,16 @@
+resource "aws_instance" "Bastion_Host" {
+  ami           = "ami-068257025f72f470d"
```
Comment (pasc-ed) [inline bastion_host.tf]: Could you use a data source instead for looking for the ami ?

File: security_groups.tf [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  name        = "server-sg"
+  description = "Allow connection for server security group."
+  vpc_id      = data.aws_vpc.vpc_group3.id
+
+  ingress {
+    description = "Allow port 22"
+    from_port   = 22
+    to_port     = 22
+    protocol    = "tcp"
+    cidr_blocks = ["110.224.205.97/32"]
```
Comment (pasc-ed) [inline security_groups.tf]: This is the security group of the ASG instances, so SSH access needs to be allowed only from the bastion host from the VPC/subnet cidr or the bastion's SG id.

File: target_groups.tf [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+resource "aws_autoscaling_attachment" "asg_attachment_blue" {
+  autoscaling_group_name = aws_autoscaling_group.blue_asg.id
+  lb_target_group_arn    = aws_lb_target_group.blue.arn
+}
+
+# for green target group
+
+resource "aws_lb_target_group" "green" {
+  # name     = "green-tg-${random_pet.app.id}-lb"
```
Comment (pasc-ed) [inline target_groups.tf]: Why is this line commented ? If it's not required, could you remove it please ?

**Review by pasc-ed — CHANGES_REQUESTED** [review]

File: autoscaling.tf (hunk 1 above)
Comment (nisha26sahu) [reply autoscaling.tf]: we have not merged that user_data file here, but we will use it..that's why.

File: autoscaling.tf (hunk 2 above)
Comment (Ghaneshkarthick) [reply autoscaling.tf]: By linking the security group of the bastion host and load balancer to the created instances,we've restricted access  for ssh only through bastion host and http connection only through load balancer. @pasc-ed

File: bastion_host.tf (hunk 3 above)
Comment (nisha26sahu) [reply bastion_host.tf]: ok, we will make the required changes. @pasc-ed

File: autoscaling.tf (hunk 2 above)
Comment (Ghaneshkarthick) [reply autoscaling.tf]: We've also negated the need for a seperate security group for the asg by doing so. @pasc-ed

Comment (pasc-ed) [reply autoscaling.tf]: @Ghaneshkarthick the restriction to bastion host and http should be inside the security group. For security, it's always best to have individual security groups for each resources without sharing rules between them.

**Pull request closed without merging**